### DISCUSSION TITLE
Dataverse host for Power Fx

### DISCUSSION THREAD

**Discussion by filcole** [microsoft/Power-Fx#1556] [General]
Currently to use Power Fx within Dataverse it's required to use the preview Dependent Assemblies functionality to make the Power Fx libraries available to the plugin.

With the increasing use of Power Fx in the Platform, and with Power Fx nearing 1.0 will we be able to use Power Fx from within a plugin without needing to use dependent assemblies?  I suspect that Microsoft desire only one instance of PowerFx libraries per Dataverse environment.

**Comment by filcole** [comment]
A specific use case is 'Power Fx' tokens, e.g. https://jonasr.app/xrm-tokens/#powerfx

**Comment by MikeStall** [comment]
I would hope so. 

we want to make it very easy for a plugin to invoke Fx within the plugin. This is effectively what Low Code plugins are doing. 

But there are 2 key challenges:
1. Fx hosting APIs are large - especially because you need to be able to control symbols and load custom APIs.  We don't necessarily want to expand IOrganizationService with all that functionality. 

2. We also want the flexibility of letting the plugin control its version of Fx.
